### PR TITLE
AN-648/hls beam issue fixed

### PR DESCRIPTION
**Pull request by jashan1498**

### CODE REVIEW

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 1]
```diff
@@ -1370,8 +1370,12 @@ class MeetingViewModel(
         hmsSDK.startHLSStreaming(config, object : HMSActionResultListener {
             override fun onError(error: HMSException) {
                 viewModelScope.launch {
-                    _events.emit(Event.Hls.HlsError(error))
-                    hlsToggleUpdateData.postValue(false)
+                    if (error.message.contains("invalid input")) {
+                        startHls(hlsUrl, recordingConfig, true)
```
Comment (PratimMallick) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt:1374]: What happens If `invalid input` error comes even after starting HLS without `null` meetingVariants? Then this is infinite loop?

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 2]
```diff
@@ -1370,8 +1370,12 @@ class MeetingViewModel(
         hmsSDK.startHLSStreaming(config, object : HMSActionResultListener {
             override fun onError(error: HMSException) {
                 viewModelScope.launch {
-                    _events.emit(Event.Hls.HlsError(error))
-                    hlsToggleUpdateData.postValue(false)
+                    if (error.message.contains("invalid input")) {
```
Comment (AniketSK) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt:1373]: Should never do string matching of errors. Only match on error codes if available.
But why are you starting hls after invalid input?

**Review by AniketSK — CHANGES_REQUESTED** [review]

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 1 above)
Comment (jashan1498) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt:1374]: actually when i checked with ravi .. its an error that only comes when destinations are null

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 2 above)
Comment (jashan1498) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt:1373]: actually its an error that is thrown when `destinations` are null , here is the web pr which i got logic from 

https://github.com/100mslive/100ms-web/blob/main/src/components/Streaming/HLSStreaming.jsx#L56

**Comment by AniketSK** [comment]
Should be fixed via sdk

**Pull request closed without merging**